pkg/credentialprovider/keyring.go

```diff
@@ -57,11 +57,11 @@ type lazyDockerKeyring struct {
 func (dk *BasicDockerKeyring) Add(cfg DockerConfig) {
 	if dk.index == nil {
 		dk.index = make([]string, 0)
-		dk.creds = make(map[string][]docker.AuthConfiguration)
+		dk.creds = make(map[string][]dockertypes.AuthConfig)
 	}
 	for loc, ident := range cfg {
 
-		creds := docker.AuthConfiguration{
+		creds := dockertypes.AuthConfig{
```

Comment:
Yeah, you should definitely not be deserializing directly from a file (or
any user input) into some docker client library's type, unless you have a
contract signed in blood that they'll never change the type. :)

On Wed, Apr 13, 2016 at 1:04 PM, David Eads <notifications@github.com>
wrote:

> In pkg/credentialprovider/keyring.go
> <https://github.com/kubernetes/kubernetes/pull/23800#discussion_r59615455>
> :
>
> >  	}
> >  	for loc, ident := range cfg {
> >
> > -		creds := docker.AuthConfiguration{
> > +		creds := dockertypes.AuthConfig{
>
> Docker uses the AuthConfig to hold the config file information (see code),
> so at least we know that it is feasible to replace our own type with
> AuthConfig. :)
>
> Hence my question to @lavalamp <https://github.com/lavalamp>. I lean
> towards owning my own data format and converting so that a dependency
> rebase will never bork my data serialization without me knowing it, but he
> owns the API-ish things for kube.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23800/files/4c318ea8f212fa1186660e369c61ad087e1d7666#r59615455>
>